apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.ts

```diff
@@ -73,6 +73,9 @@ async function ensureUpToDate(locals: Record<string, any>) {
 
 async function processFileUpload(req: Request, res: Response) {
   if (!res.locals.assessment_instance.open) throw new Error('Assessment is not open');
+  if (!res.locals.assessment.allow_personal_notes) {
+    throw new Error('This assessment does not allow personal notes.');
```

Comment:
Done (including the other errors)

Reply:
Just noting here that one of the existing tests breaks because of this change as it expects a 500 error code. At least my interpretation is that 403 is more reasonable for trying to attach a file to a non-open assessment, but I'm also fine reverting that change and the test case update I committed for now.

Reply:
A 403 is without a doubt the correct status code to use here. Thanks for updating the test as well.